pkgs/native_assets_builder/test/build_runner/concurrency_shared_test_helper.dart

```diff
@@ -23,23 +23,27 @@ void main(List<String> args) async {
     dartExecutable: dartExecutable,
     fileSystem: const LocalFileSystem(),
   ).build(
-    // Set up the code config, so that the builds for different targets are
+    // Set up the code input, so that the builds for different targets are
     // in different directories.
-    configCreator: () => BuildConfigBuilder()
-      ..setupCodeConfig(
+    inputCreator: () => BuildInputBuilder()
+      ..config.setupShared(buildAssetTypes: [
+        CodeAsset.type,
```

Comment:
Before it was `DataAsset` only, why did you have to add `CodeAsset` here?

Reply:
We shouldn't be calling `setupCode` below without the `CodeAsset.type` here.
We need the `setupCode` to ensure differences in the directories.

I guess it's technically correct to add more config that's never read by anyone, but it's messy.

Reply:
Maybe the `setupCode` call should be removed instead if it worked with data assets only?

There's 3 things that should be in sync
* the `buildAssetTypes`
* the config
* the validation

With your changes we support code assets but it doesn't validate them, see
```
    buildValidator: (input, output) async =>
        await validateDataAssetBuildOutput(input, output),
```

Reply:
Adding the code validator